### PR TITLE
Matrix improvements

### PR DESCRIPTION
**Pull request by andrevmatos**
Add `account_data` support to `GMatrixClient` and Rooms. This suport allows using `(GMatrixClient|Room).(set|get)_account_data` to store/retrieve a dict of key-values to and from global account data and per-room account data.

### CODE REVIEW

File: raiden_libs/network/matrix/client.py [hunk 1]
```diff
@@ -36,9 +39,12 @@ def __init__(
         )
         self.should_listen = False
         self.sync_thread = None
-        self.greenlets = list()  # type: List[gevent.Greenlet]
+        self.greenlets: List[gevent.Greenlet] = list()
         self.api.session.headers.update({'Connection': 'close'})
 
+        # locks each account_data's key until it's _sync'ed
+        self.account_data_locks: Dict[str, gevent.lock.Semaphore] = dict()
```
Comment (ulope) [inline raiden_libs/network/matrix/client.py]: Wouldn't it make more sense to use `gevent.Event()` here? Seems to me like you're already using the semaphores like events already.

Also using a `defaultdict` could make the `set_|get_account_data()` a bit easier.

File: raiden_libs/network/matrix/room.py [hunk 2]
```diff
@@ -16,6 +18,11 @@ def __init__(self, client, room_id):
         super().__init__(client, room_id)
         self._members = {}
 
+        # dict of 'type': 'content' key/value pairs
+        self.account_data: Dict[str, Dict[str, Any]] = dict()
+        # locks each account_data's key until it's _sync'ed
+        self.account_data_locks: Dict[str, gevent.lock.Semaphore] = dict()
```
Comment (ulope) [inline raiden_libs/network/matrix/room.py]: Same as above

File: raiden_libs/network/matrix/client.py (hunk 1 above)
Comment (andrevmatos) [reply raiden_libs/network/matrix/client.py]: I thought about it, but that may not work or be less expresive, as the intent here is to make `set_account_data` block and wait until we get the same data we set on the next `sync` loop. Using event can make this notification happen, but won't help blocking in the case a concurrent call to `set_account_data` was made, which is the main purpose of this locking mechanism.

Comment (andrevmatos) [reply raiden_libs/network/matrix/client.py]: Notice also that the `set_account_data` should trigger an event immediatelly, so the `sync` loop should happen right after (not in 30s only or so), so it's pretty quick and just a race condition safe net.

**Pull request merged**